src/main.c

```diff
@@ -102,12 +118,18 @@ void setup(void)
     fwdgt_window_value_config(0x0FFF);
     fwdgt_enable(); // Enable watchdog
     status_leds_init();
+    button_init();
 
     /* Reset WD */
     fwdgt_counter_reload();
 
     int baudrate = check_bootloader_data();
 
+#ifdef BUTTON
+    if (gpio_in_read(buttonPin) == BUTTON_PRESSED)
```

Comment:
Should this take debounce into account?
...or is there a cap on the PCB (I cannot remember...)?

Reply:
What are you thinking? A 50ms loop checking for BUTTON_RELEASED to prevent accidentally entering BL due to noise?

Reply:
added

Reply:
sorry for the late reply!
Yes, not only the noise but a  press / release makes signal pretty ugly.